src/key-value-pairs/__tests__/key-value-pairs.test.tsx

```diff
@@ -0,0 +1,150 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import * as React from 'react';
+import { render } from '@testing-library/react';
+import createWrapper from '../../../lib/components/test-utils/dom';
+import KeyValuePairs from '../../../lib/components/key-value-pairs';
+
+function renderKeyValuePairs(jsx: React.ReactElement) {
+  const { container } = render(jsx);
+  return { wrapper: createWrapper(container).findKeyValuePairs()! };
+}
+
+describe('KeyValuePairs', () => {
+  describe('item rendering', () => {
+    test('renders correctly', () => {
+      const { wrapper } = renderKeyValuePairs(
+        <KeyValuePairs
+          items={[
+            {
+              label: 'Label for key',
+              value: 'Value',
+            },
+          ]}
+        />
+      );
+
+      expect(wrapper.findItems()[0]!.findLabel()!.getElement()).toHaveTextContent('Label for key');
+      expect(wrapper.findItems()[0]!.findValue()!.getElement()).toHaveTextContent('Value');
+    });
+
+    test('renders label with info correctly', () => {
+      const { wrapper } = renderKeyValuePairs(
+        <KeyValuePairs
+          items={[
+            {
+              label: 'Label for key',
+              value: 'Value',
+              info: 'Info',
+            },
+          ]}
+        />
+      );
+
+      expect(wrapper.findItems()[0]!.findInfo()!.getElement()).toHaveTextContent('Info');
```

Comment:
would be good to add a check here for info auto-labelling

Reply:
added